### ISSUE TITLE
Morrigan server .start method promise resolves before server is ready

### ISSUE THREAD

**Issue by Adicitus** [Adicitus/morrigan#2]
This will fail to stop the server if the .setup/.start methods take too long to finish:
```
const Morrigan = require('@adicitus/morrigan.server')
let server = new Morrigan({ http: { secure: false, port: 80 }, database: { connectionString: "mongodb://127.0.0.1:27017", dbname: 'morrigan-server-test' }, logger: { level: 'silly', logDir: `${__dirname}/data/log` } })

await server.start()
await server.stop()
```

For the sake of automation and testing, this should be enough to cycle through all stages of the server lifecycle: 
`let server = new Morrigan(/*settings here*/); await server.start(); await server.stop()`

**Comment by Adicitus** [comment]
The issue can be worked around using a promise:
```
let p = new Promise(resolve => server.on('ready', resolve))
server.start()
await p
server.stop()
```

**Comment by Adicitus** [comment]
This should be fixed in v0.9.5.
